### ISSUE TITLE
Add convenient way to use xidlehook with xss-lock

### ISSUE THREAD

**Issue by wchresta** [nix-community/home-manager#3306]
### Description

`services.screen-locker` provides a convenient way to define the `xss-lock` service together with the `xautolock` service to get automatic screenlock after idle. Unfortunately, `xautolock` lacks some features that `xidlehook` would provide.

While there is a definition for xidlehook in `services.xidlehook`, there is no convenient way to use xidlehook together with xss-lock. So, to achieve screen locking via xss-lock and xidlehook, one has to define their own xss-lock service definition.

This FR asks to add a convenient way to use xidlehook together with xss-lock without the user having to define either service themselves.

This could be achieved with one of the following:
* Add xidlehook as an alternative to xautolock in screen-locker
* Add a convenient way to disable *both* xautolocker and xset in screen-locker, such that one can use xidlehook instead via `services.xidlehook`
* Refactor screen-locker and pull out `services.xss-lock` such that it can be used in combination with `services.xidlehook`.

**Comment by Aleksanaa** [comment]
Indeed, xss-lock should not be bound to any specific implementation of auto locking. It can even be used in wayland (but I just discovered swayidle has the same feature to listen to systemd event).
